tests/std/tests/GH_002030_asan_annotate_vector/test.cpp

```diff
@@ -36,16 +111,16 @@ struct custom_test_allocator {
     using is_always_equal                        = Stateless;
 };
 
-template <class T, class Pocma, class Stateless>
-constexpr bool operator==(const custom_test_allocator<T, Pocma, Stateless>& lhs,
-    const custom_test_allocator<T, Pocma, Stateless>& rhs) noexcept {
-    return Stateless::value || (&lhs == &rhs);
+template <class T1, class T2, class Pocma, class Stateless>
+constexpr bool operator==(
+    const custom_test_allocator<T1, Pocma, Stateless>&, const custom_test_allocator<T2, Pocma, Stateless>&) noexcept {
+    return Stateless::value;

```

Comment:
Does it matter if this allocator isn't a real allocator? 😛 
For real though I'm not totally sure this matters. I just want to have some test coverage for when the allocators that are passed into vector don't compare equal.

Reply:
If it works for what you want from the test - and I think it does - we can get away with a "purely for testing: this doesn't actually meet the allocator requirements" comment. The problem we want to avoid is someone naively copying this into another test because they need an allocator and think this is one.